web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,248 @@ use Drupal\taxonomy\Entity\Term;
+        $importedNode = $answer->importNode($element, TRUE);
+        $answer->documentElement->childNodes[0]->appendChild($importedNode);
+      }
+    }
+
+    $answer = faq_answer_inner_html($answer);
+    if ($question && $answer) {
+      $faq[] = (object) ['question' => $question, 'answer' => $answer];
+    }
+  }
+  $variables["dom"] = $dom;
+  return $faq;
 }
 
+/**
+ * faq_process_components gets components from drupal paragraph embeds
+ * and makes their content compatible with FAQ rich results
+ */
+function faq_process_components(&$dom) {
+  // Iterate through embedded paragraphs
+  $embeddedParagraphElements = $dom->getElementsByTagName('drupal-paragraph');
+  // Loop backwards to allow for removing elements along the way without losing track of our position
+  for ($n = $embeddedParagraphElements->length - 1; $n >= 0; --$n) {
+    $element = $embeddedParagraphElements->item($n);
+    // Get the component
+    $uuid = $element->getAttribute('data-paragraph-id');
+    $embedded_paragraph = \Drupal::service('entity.repository')->loadEntityByUuid('embedded_paragraphs', $uuid);
+    $component = $embedded_paragraph->get('paragraph')[0]->entity;
+
+    // Process component by its type
+    $componentType = $component->getType();
+    if ($componentType == "uswds_accordion") {
+
+      // iterate through accordion panels
+      foreach ($component->get('field_accordion_section') as $accordionPanel) {
+        // Get the accordion panel's node
+        $accordionPanelTID = $accordionPanel->get('target_id')->getValue();
+        $accorionPanelNode = \Drupal::entityTypeManager()->getStorage('paragraph')->load($accordionPanelTID);
+
+        // Get this panel's button text and add it to the dom as an h2
+        $accorionPanelButtonText = $accorionPanelNode->get('field_accordion_section_title')[0]->value;
+        $accorionHeading = $dom->createElement('h2');
+        $accorionHeading->textContent = $accorionPanelButtonText;
+        $element->parentNode->insertBefore($accorionHeading, $element);
+
+        // Get this panel's body content
+        $accorionPanelBodyNodeID = $accorionPanelNode->get('field_accordion_section_body')[0]->get('target_id')->getValue();
+        $accorionPanelBodyNode = \Drupal::entityTypeManager()->getStorage('paragraph')->load($accorionPanelBodyNodeID);
+
+        // append accordion body content
+        $accorionBody = new DOMDocument();
+        $accorionBody->loadHTML($accorionPanelBodyNode->get('field_text')[0]->value);
+        foreach ($accorionBody->documentElement->childNodes[0]->childNodes as $accorionBodyElement) {
+          $accorionBodyElement = $dom->importNode($accorionBodyElement, TRUE);
+          if ($accorionBodyElement) {
+            $element->parentNode->insertBefore($accorionBodyElement, $element);
+          }
+        }
+      }
+    }
+    elseif ($componentType == "uswds_alert" && $component->get('field_alert_status')[0]->value == NULL) {
```

Comment:
Let's add a null-safe parameter here too. Maybe more than one, but I'm specifically getting this warning: 

`Warning: Attempt to read property "value" on null in faq_process_components() (line 178 of /var/www/web/themes/custom/usagov/usagov.theme)`

Reply:
Good call